src/use_case/preferences/PreferencesInteractor.java

```diff
@@ -1,6 +1,32 @@
 package src.use_case.preferences;
 import src.entity.User;
 import java.util.*;
-public class PreferencesInteractor {
+public class PreferencesInteractor implements PreferencesInputBoundary{
+    final PreferencesUserDataAccessInterface userDataAccessObject;
+    final PreferencesOutputBoundary preferencesPresenter;
 
+    public PreferencesInteractor(PreferencesUserDataAccessInterface userDataAccessInterface,
+                                PreferencesOutputBoundary preferencesOutputBoundary){
+        this.userDataAccessObject = userDataAccessInterface;
+        this.preferencesPresenter = preferencesOutputBoundary;
+    }
+
+
+    @Override
+    public void execute(PreferencesInputData preferencesInputData){
+        int userId = preferencesInputData.getUserId();
+        HashMap<String, Boolean> dietary = preferencesInputData.getDietary();
+        HashMap<String, String> conditions = preferencesInputData.getConditions();
+        HashMap<String, Boolean> allergies = preferencesInputData.getAllergies();
+        int userID = preferencesInputData.getUserId();
+        if (!userDataAccessObject.existByUserID(userID)){
+            preferencesPresenter.prepareFailView("Unable to add preferences as user does not exist.Please try again.");
+        } else {
+            userDataAccessObject.savePreferences(userId, dietary, allergies, conditions);
```

Comment:
I think you can now add an if statement check here when calling userDataAccessObject.savePreferences since it will return a boolean value now. Something like if (userDataAccessObject.savePreferences(.......)) and then initiate the prepareSuccessView or else call in the prepareFailView.